metadata/README.md

```diff
@@ -0,0 +1,15 @@
+## Metadata 
```

Comment:
I think that it would be beneficial to write down the contract between metadata generation and runtime pallets. What is enforced by the tool in order for it to work properly (f.e pallet names etc)